tempus_fugit_minecraft/main.py

```diff
@@ -744,9 +744,9 @@ def on_key_press(self, symbol, modifiers):
             else:
                 self.resume_game()
             self.paused = not self.paused
-        elif symbol == key.TAB:
+        elif symbol == key.TAB and not self.paused:
             self.flying = not self.flying
-        elif symbol in self.num_keys:
+        elif symbol in self.num_keys and not self.paused:
```

Comment:
If you apply the suggestion from my previous comment about splitting the escape key handling, these changes won't be needed.